examples/DocQA/app.py

```diff
@@ -0,0 +1,300 @@
+import asyncio
+import json
+import os
+import uuid
+from queue import Queue
+from threading import Thread
+from typing import AsyncGenerator, Generator, List, Optional
+
+import chromadb
+import gradio as gr
+import requests
+from chromadb.utils import embedding_functions
+from dotenv import load_dotenv
+from llama_stack_client import LlamaStackClient
+from llama_stack_client.lib.agents.agent import Agent
+from llama_stack_client.lib.agents.event_logger import EventLogger
+from llama_stack_client.types.agent_create_params import AgentConfig
+from llama_stack_client.types.memory_insert_params import Document
+import re
+
+
+# Load environment variables
+load_dotenv()
+
+HOST = os.getenv("HOST", "localhost")
+PORT = int(os.getenv("PORT", "5000"))
+GRADIO_SERVER_PORT = int(os.getenv("GRADIO_SERVER_PORT", "7861"))
+USE_GPU = os.getenv("USE_GPU", False)
+MODEL_NAME = os.getenv("MODEL_NAME", "meta-llama/Llama-3.2-1B-Instruct")
+# if use_gpu, then the documents will be processed to output folder
+DOCS_DIR = "/root/rag_data/output" if USE_GPU else "/root/rag_data/"
```

Comment:
why is this different depending on whether we use a GPU or not?

Reply:
Our ingest pipeline will take the `/root/rag_data/` data folder and save results to `/root/rag_data/output` when using GPU. Otherwise the data folder is just `/root/rag_data/`

Reply:
@wukaixingxp still don't know why the output folder is _different_ when using GPU vs. not? if you aren't using GPU, where does the ingest pipeline save the results?

Reply:
ingest pipeline is just for image: basically it takes everything in `/root/rag_data`, find out if there is any image embed in the PDF, split the image and use 11B to generate image description, then it will save the original text with image description into `/root/rag_data` folder so that everything is now text data, ready to be used by 3B RAG agent. Running 11B on CPU is sooo slow and we have not enable this in the current stage. We believe this can be a P1 feature to have. Now we only support text data and will ignore embedded images, thus just take everything in the `/root/rag_data` folder